docs/dev_guide/extending_fido.rst

```diff
@@ -0,0 +1,164 @@
+
+* **Converters**: Adding a converter to the walker, adds the function to the both the creator and the applier.
+  For the VSO client this is used to convert each supported attr into a `~sunpy.net.attr.ValueAttr` which is then later processed by the appliers and creators.
+  This pattern can be useful if you would otherwise have to repeat a lot of logic in each of the applier functions for each type of Attr you support.
+
+An Example of ``AttrWalker``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this example we will write a parser for some simple queries which uses `~sunpy.net.attr.AttrWalker` to convert the query to a `dict` of URL query parameters for a HTTP GET request.
+Let's imagine we have a web service which you can do a HTTP GET request to ``https://sfsi.sunpy.org/search`` for some imaginary data from an instrument called SFSI (SunPy Fake Solar Instrument).
+This GET request takes three query parameters ``startTime``, ``endTime`` and ``level``, so a request might look something like: ``https://sfsi.sunpy.org/search?startTime=2020-01-02T00:00:00&endTime=2020-01-02T00:00:00&level=1``.
+Which would search for level one data between 2020-01-01 and 2020-01-02.
+
+As `~sunpy.net.attrs` has `~sunpy.net.attrs.Time` and `~sunpy.net.attrs.Level` we don't need to define any of our own attrs for this client.
+We do however want to write our own walker to convert them to the form out client's ``search()`` method wants to send them to the server.
+
+The first step is to setup the walker and define a creator method which will return either a dict (for a single query) or a list of dicts for multiple queries.
+
+..code-block:: python
+
+    import sunpy.net.atrrs as a
+    from sunpy.net.attr import AttrWalker, AttrAnd, AttrOr, DataAttr
+
+    walker = AttrWalker()
+
+    @walker.add_creator(AttrOr)
+    def create_or(wlk, tree):
+        results = []
+        for sub in tree.attrs:
+            results.append(wlk.create(sub))
+
+        return results
+
+    @walker.add_creator(AttrAnd, DataAttr)
+    def create_and(wlk, tree):
+        result = dict()
+        wlk.apply(tree, result)
+        return result
+
+Adding "Attrs" to Registry
+##########################
+
+Registering of "attrs" ensures discoverability of search attributes supported by the corresponding SunPy Client.
+For adding them to the Registry, we need to define a ``classmethod`` :meth:`~sunpy.net.base_client.BaseClient.register_values` that returns a dictionary of registered values.
+This dictionary should have `~sunpy.net.attr.Attr` classes as keys and a list of tuples corresponding to that key representing the possible values key "attr" can take.
+Each tuple comprises of two elements. The first one is value and the second element contains a brief description of that value.
+An example of writing ``register_values()`` for `~sunpy.net.dataretriever.GenericClient` is provided above. Please note that it can be defined in a similar way for full clients too.
```

Comment:
```suggestion
An example of writing ``register_values()`` for `~sunpy.net.dataretriever.GenericClient` is provided above.
Please note that it can be defined in a similar way for full clients too.
```

Reply:
Done.